collection.go

```diff
@@ -379,18 +391,25 @@ type collectionLoader struct {
 	handles  *handleCache
 }
 
-func newCollectionLoader(coll *CollectionSpec, opts *CollectionOptions) *collectionLoader {
+func newCollectionLoader(coll *CollectionSpec, opts *CollectionOptions) (*collectionLoader, error) {
 	if opts == nil {
 		opts = &CollectionOptions{}
 	}
 
+	// check that there exist maps in coll.Maps for all maps in opts.OverrideMaps
+	for name := range opts.OverrideMaps {
+		if _, ok := coll.Maps[name]; !ok {
+			return nil, fmt.Errorf("override map %s doesn't exist in the spec", name)
```

Comment:
Going with the 'replacement' term:
```suggestion
			return nil, fmt.Errorf("replacement map %s not found CollectionSpec", name)
```